hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteFunction.java

```diff
@@ -453,7 +454,7 @@ private boolean flushBucket(DataBucket bucket) {
         .endInput(false)
         .build();
 
-    this.eventGateway.sendEventToCoordinator(event);
+    this.eventGateway.sendEventToCoordinator(new WriteResultEvent(event, currentInstant));
```

Comment:
`WriteMetadataEvent.instantTime` is the the instant subtask written into.  
`WriteResultEvent.curInstant` is the pending instant in timeline when sending it to JM, it will get update when task restart.  
So they maybe different after task restart.